docs/guide/configuration.md

```diff
@@ -0,0 +1,119 @@
+# Configuration
+
+The Next Gen UI Agent can be configured in two ways: `programmatically` using Python dictionaries or `declaratively` using YAML configuration files.
+
+## Configuration Options
+
+### `inference` [`InferenceBase`, required]
+The LLM inference engine to use for component selection and configuration.
+
+The `inference` parameter accepts any class that extends `InferenceBase`. The Next Gen UI Agent provides the following built-in inference implementations:
+
+- `next_gen_ui_agent.model.LangChainModelInference`: Wrapper for LangChain-compatible language models. Supports any LangChain LLM implementation including ChatOllama, ChatOpenAI, ChatAnthropic, and other LangChain model providers.
+- `next_gen_ui_llama_stack.LlamaStackAgentInference`: Integration with Standalone Llama Stack
+- `next_gen_ui_llama_stack_embedded.LlamaStackEmbeddedAsyncAgentInference`: Integration with Embedded Llama Stack
+
+
+### `component_system` [`str`, optional]
+
+Component system for rendering (default: `"json"`)
+
+### `unsupported_components` [`bool`, optional]
+
+Whether to allow unsupported UI components (default: `False`)
+
+### `component_selection_strategy` [`str`, optional]
+
+Strategy for component selection (default: `"default"`)
```

Comment:
Name all the available strategies here plase

Reply:
ah. I deleted what was there.